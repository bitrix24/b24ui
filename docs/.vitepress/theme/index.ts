/**
 * This is custom theme
 * We change some css
 * @see node_modules/vitepress/dist/client/theme-default/index.js
 */
import { ref, watch, toValue } from 'vue'
import { createRouter, createMemoryHistory } from 'vue-router'
import bitrix24UIPlugin from '@bitrix24/b24ui-nuxt/vue-plugin'
import Layout from './components/ui/Layout.vue'
import ComponentShowExample from './components/ui/ComponentShowExample.vue'
import ComponentProps from './components/ui/ComponentProps.vue'
import ComponentSlots from './components/ui/ComponentSlots.vue'
import ComponentEmits from './components/ui/ComponentEmits.vue'
import Description from './components/ui/Description.vue'
import Placeholder from './components/ui/Placeholder.vue'
import VPBadge from 'vitepress/dist/client/theme-default/components/VPBadge.vue'
import 'vitepress/dist/client/theme-default/styles/fonts.css'
import './tailwind.css'
// region ory theme ////
import './styles/components/custom-block.css'
import './styles/components/vp-code.css'
import './styles/components/vp-code-group.css'
import './styles/components/vp-doc.css'
import './styles/components/vp-sponsor.css'
// endregion ////
import './theme.css'

globalThis.useFetch = async (url: string, options: RequestInit & { params?: any, transform?: (data) => any } = {}) => {
  const data = ref()
  const status = ref('idle')
  async function _fetch() {
    status.value = 'loading'
    try {
      const params = options.params
        ? Object.fromEntries(
            Object.entries(options.params).map(([key, value]) => [key, toValue(value)])
          )
        : {}

      const queryString = new URLSearchParams(params as any).toString()
      const fullUrl = queryString ? `${url}?${queryString}` : url

      data.value = await fetch(fullUrl, options).then(r => r.json()).then(r => options.transform ? options.transform(r) : r)
      status.value = 'success'
    } catch (error) {
      console.error(error)
      status.value = 'error'
    }
  }
  _fetch()

  if (options?.params?.q) {
    watch(
      options.params.q,
      () => {
        _fetch()
      },
      { deep: true }
    )
  }

  return Promise.resolve({
    data,
    status
  })
}

export default {
  Layout,
  enhanceApp: async ({ app }) => {
    // region components ////
    app.component('Badge', VPBadge)
      .component('Description', Description)
      .component('ComponentShowExample', ComponentShowExample)
      .component('ComponentProps', ComponentProps)
      .component('ComponentSlots', ComponentSlots)
      .component('ComponentEmits', ComponentEmits)
      .component('Placeholder', Placeholder)
    // endregion ////

    if (!import.meta.env.SSR) {
      const router = createRouter({
        routes: [
          // { path: '', component: app }
        ],
        history: createMemoryHistory()
        // history: createWebHistory()
      })
      app.use(router)

      app.use(bitrix24UIPlugin)
    }
  }
}
