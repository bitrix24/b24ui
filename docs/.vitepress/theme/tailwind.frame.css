/*
@plugin '@bitrix24/b24style';
@plugin "@bitrix24/b24ui-nuxt";
*/
@import "tailwindcss" source("../../");
@import "@bitrix24/b24ui-nuxt";
/*
*/



