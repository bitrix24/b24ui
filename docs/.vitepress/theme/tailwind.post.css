@import "tailwindcss" source("../../../docs");
@import "@bitrix24/b24ui-nuxt";

/**
 * bxshef: Custom
 * -------------------------------------------------------------------------- */

:root
{
}

.dark
{
  --vp-c-bg: #0f172a;
  --vp-c-bg-alt: #161618;
  --vp-c-bg-elv: #202127;
  --vp-c-bg-soft: #202127;
}

.bg-grid-example {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

.dark\:bg-grid-example:is(.dark *) {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}
