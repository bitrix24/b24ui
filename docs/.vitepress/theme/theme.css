/**
 * bxshef: Custom
 * -------------------------------------------------------------------------- */

:root
{
  --sh-scrollbar-thumb: #3c3c43;
  --sh-scrollbar-background: #ebebef;
}

.dark
{
  --vp-c-bg: #0f172a;
  --vp-c-bg-alt: #161618;
  --vp-c-bg-elv: #202127;
  --vp-c-bg-soft: #202127;
  --vp-code-block-bg: #1a2234;

  --sh-scrollbar-thumb: #bdc1c6;
  --sh-scrollbar-background: #3c3c43;
}

.bg-grid-example {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

.dark\:bg-grid-example:is(.dark *) {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

iframe
{
  min-width: 100%;
  border: none;
  scrollbar-width: thin;
  background-color: transparent;
}

/** region scrollbar */
* {
  scrollbar-color: var(--sh-scrollbar-thumb) var(--sh-scrollbar-background);
  scrollbar-width: thin;
}

/* Styling the scrollbar in Chrome and Safari */
*::-webkit-scrollbar {
  width: 6px;
}

/* Set thumb color for Chrome and Safari */
*::-webkit-scrollbar-thumb {
  background-color: var(--sh-scrollbar-thumb);
  border-radius: 4px;
}

/* Set track color for Chrome and Safari */
*::-webkit-scrollbar-track {
  background-color: var(--sh-scrollbar-background);
}
/** endregion */
