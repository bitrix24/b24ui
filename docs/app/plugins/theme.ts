export default defineNuxtPlugin({
  enforce: 'post',
  setup() {
  }
})
