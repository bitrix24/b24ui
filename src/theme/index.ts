export { default as advice } from './advice'
export { default as avatar } from './avatar'
export { default as avatarGroup } from './avatar-group'
export { default as badge } from './badge'
export { default as button } from './button'
export { default as buttonGroup } from './button-group'
export { default as checkbox } from './checkbox'
export { default as chip } from './chip'
export { default as container } from './container'
export { default as countdown } from './countdown'
export { default as formField } from './form-field'
export { default as input } from './input'
export { default as kbd } from './kbd'
export { default as link } from './link'
export { default as progress } from './progress'
export { default as radioGroup } from './radio-group'
export { default as range } from './range'
export { default as separator } from './separator'
export { default as skeleton } from './skeleton'
export { default as switch } from './switch'
export { default as textarea } from './textarea'
export { default as tooltip } from './tooltip'
