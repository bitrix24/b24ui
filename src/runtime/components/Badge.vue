<script lang="ts">
import type { AppConfig } from '@nuxt/schema'
import theme from '#build/b24ui/badge'
import type { UseComponentIconsProps } from '../composables/useComponentIcons'
import type { AvatarProps } from '../types'
import type { ComponentConfig } from '../types/utils'

type Badge = ComponentConfig<typeof theme, AppConfig, 'badge'>

export interface BadgeProps extends Omit<UseComponentIconsProps, 'loading' | 'loadingIcon'> {
  /**
   * The element or component this component should render as.
   * @defaultValue 'span'
   */
  as?: any
  label?: string | number
  /**
   * @defaultValue 'default'
   */
  color?: Badge['variants']['color']
  /**
   * @defaultValue 'normal'
   */
  depth?: Badge['variants']['depth']
  /**
   * @defaultValue 'md'
   */
  size?: Badge['variants']['size']
  /**
   * Shows 'underline' on hover
   * @defaultValue false
   */
  useLink?: boolean
  /**
   * Shows icons.close on the right side
   * @defaultValue false
   */
  useClose?: boolean
  onCloseClick?: ((event: MouseEvent) => void | Promise<void>) | Array<((event: MouseEvent) => void | Promise<void>)>
  /**
   * Fills the background
   * @defaultValue false
   */
  useFill?: boolean
  class?: any
  b24ui?: Badge['slots']
}

export interface BadgeSlots {
  leading(props?: {}): any
  default(props?: {}): any
  trailing(props?: {}): any
}
</script>

<script setup lang="ts">
import { computed } from 'vue'
import { Primitive } from 'reka-ui'
import { useAppConfig } from '#imports'
import { useButtonGroup } from '../composables/useButtonGroup'
import { useComponentIcons } from '../composables/useComponentIcons'
import { tv } from '../utils/tv'
import Cross20Icon from '@bitrix24/b24icons-vue/actions/Cross20Icon'
import B24Avatar from './Avatar.vue'

const props = withDefaults(defineProps<BadgeProps>(), {
  as: 'span'
})

defineSlots<BadgeSlots>()

const appConfig = useAppConfig() as Badge['AppConfig']

async function onCloseClickWrapper(event: MouseEvent) {
  const callbacks = Array.isArray(props.onCloseClick) ? props.onCloseClick : [props.onCloseClick]
  try {
    await Promise.all(callbacks.map(fn => fn?.(event)))
  } finally { /* empty */ }
}

const { orientation, size: buttonGroupSize } = useButtonGroup<BadgeProps>(props)
const { isLeading, leadingIconName } = useComponentIcons(props)

const b24ui = computed(() => tv({ extend: tv(theme), ...(appConfig.b24ui?.badge || {}) })({
  color: props.color,
  depth: props.depth,
  size: buttonGroupSize.value || props.size,
  buttonGroup: orientation.value,
  useLink: Boolean(props.useLink),
  useClose: Boolean(props.useClose),
  useFill: Boolean(props.useFill),
  leading: Boolean(isLeading.value)
}))
</script>

<template>
  <Primitive
    :as="as"
    :class="b24ui.base({ class: [props.b24ui?.base, props.class] })"
  >
    <Primitive
      :as="as"
      :class="b24ui.wrapper({ class: props.b24ui?.wrapper })"
    >
      <slot name="leading">
        <Component
          :is="leadingIconName"
          v-if="isLeading && leadingIconName"
          :class="b24ui.leadingIcon({ class: props.b24ui?.leadingIcon })"
        />
        <B24Avatar
          v-else-if="!!avatar"
          :size="((props.b24ui?.leadingAvatarSize || b24ui.leadingAvatarSize()) as AvatarProps['size'])"
          v-bind="avatar"
          :class="b24ui.leadingAvatar({ class: props.b24ui?.leadingAvatar })"
        />
      </slot>

      <span :class="b24ui.label({ class: props.b24ui?.label })">
        <slot>
          <span v-if="label">
            {{ label }}
          </span>
        </slot>
      </span>
    </Primitive>
    <slot name="trailing">
      <Cross20Icon
        v-if="useClose"
        :class="b24ui.trailingIcon({ class: props.b24ui?.trailingIcon })"
        aria-hidden="true"
        @click.stop.prevent="onCloseClickWrapper"
      />
    </slot>
  </Primitive>
</template>
