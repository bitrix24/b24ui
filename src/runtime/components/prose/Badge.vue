<script lang="ts">
import type { AppConfig } from '@nuxt/schema'
import theme from '#build/b24ui/prose/badge'
import type { ComponentConfig } from '../../types/tv'

type ProseBadge = ComponentConfig<typeof theme, AppConfig, 'badge', 'b24ui.prose'>

export interface ProseBadgeProps {
  class?: any
}

export interface ProseBadgeSlots {
  default(props?: {}): any
}
</script>

<script setup lang="ts">
import { computed } from 'vue'
import { useAppConfig } from '#imports'
import { tv } from '../../utils/tv'
import B24Badge from '../Badge.vue'

const props = defineProps<ProseBadgeProps>()
defineSlots<ProseBadgeSlots>()

const appConfig = useAppConfig() as ProseBadge['AppConfig']

const b24ui = computed(() => tv({ extend: tv(theme), ...(appConfig.b24ui?.prose?.badge || {}) }))
</script>

<template>
  <B24Badge color="air-primary" :class="b24ui({ class: props.class })">
    <slot mdc-unwrap="p" />
  </B24Badge>
</template>
