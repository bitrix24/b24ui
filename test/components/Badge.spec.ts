import { describe, it, expect } from 'vitest'
import Badge, { type BadgeProps, type BadgeSlots } from '../../src/runtime/components/Badge.vue'
import ComponentRender from '../component-render'
import theme from '#build/b24ui/badge'
import SignIcon from '@bitrix24/b24icons-vue/main/SignIcon'

describe('Badge', () => {
  const sizes = Object.keys(theme.variants.size) as any
  const depths = Object.keys(theme.variants.depth) as any

  it.each([
    // Props
    ['with label', { props: { label: 'Badge' } }],
    ...sizes.map((size: string) => [`with size ${size}`, { props: { label: 'Badge', size } }]),
    ...depths.map((depth: string) => [`with primary depth ${depth}`, { props: { label: 'Badge', depth } }]),
    ...depths.map((depth: string) => [`with success depth ${depth}`, { props: { label: 'Badge', depth, color: 'success' } }]),
    ['with icon', { props: { icon: SignIcon } }],
    ['with leading and icon', { props: { leading: true, icon: SignIcon } }],
    ['with leadingIcon', { props: { leadingIcon: SignIcon } }],
    ['with trailing and icon', { props: { trailing: true, icon: SignIcon } }],
    ['with trailingIcon', { props: { trailingIcon: SignIcon } }],
    ['with avatar', { props: { avatar: { src: 'https://github.com/bitrix24.png' } } }],
    ['with avatar and leadingIcon', { props: { avatar: { src: 'https://github.com/bitrix24.png' }, leadingIcon: SignIcon } }],
    ['with avatar and trailingIcon', { props: { avatar: { src: 'https://github.com/bitrix24.png' }, trailingIcon: SignIcon } }],
    ['with square', { props: { label: 'Badge', square: true } }],
    ['with as', { props: { label: 'Badge', as: 'div' } }],
    ['with class', { props: { label: 'Badge', class: 'rounded-full font-bold' } }],
    ['with b24ui', { props: { label: 'Badge', b24ui: { label: 'font-bold' } } }],
    // Slots
    ['with default slot', { slots: { default: () => 'Default slot' } }],
    ['with leading slot', { slots: { leading: () => 'Leading slot' } }],
    ['with trailing slot', { slots: { trailing: () => 'Trailing slot' } }]
  ])('renders %s correctly', async (nameOrHtml: string, options: { props?: BadgeProps, slots?: Partial<BadgeSlots> }) => {
    const html = await ComponentRender(nameOrHtml, options, Badge)
    expect(html).toMatchSnapshot()
  })
})
