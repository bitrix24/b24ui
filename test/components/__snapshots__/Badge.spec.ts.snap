// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > renders with as correctly 1`] = `
"<div class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent">
  <div class="inline-flex items-center h-[17px] gap-1">
    <!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis underline-offset-1">Badge</span>
  </div>
  <!--v-if-->
</div>"
`;

exports[`Badge > renders with avatar and leadingIcon correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md ps-2xs font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent" leadingicon="i-lucide-arrow-left"><span class="inline-flex items-center h-[17px] gap-1"><span class="inline-flex items-center justify-center select-none overflow-hidden rounded-full align-middle bg-base-100 dark:bg-base-900 size-xs2 text-4xs font-regular shrink-0"><img role="img" src="https://github.com/benjamincanac.png" width="10" height="10" class="h-full w-full rounded-[inherit] object-cover" style="display: none;"><span class="font-medium text-base-500 dark:text-base-600 truncate"> </span></span>
<!--v-if--></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with avatar and trailingIcon correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md ps-2xs font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><span class="inline-flex items-center justify-center select-none overflow-hidden rounded-full align-middle bg-base-100 dark:bg-base-900 size-xs2 text-4xs font-regular shrink-0"><img role="img" src="https://github.com/benjamincanac.png" width="10" height="10" class="h-full w-full rounded-[inherit] object-cover" style="display: none;"><span class="font-medium text-base-500 dark:text-base-600 truncate"> </span></span>
<!--v-if--></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with avatar correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md ps-2xs font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><span class="inline-flex items-center justify-center select-none overflow-hidden rounded-full align-middle bg-base-100 dark:bg-base-900 size-xs2 text-4xs font-regular shrink-0"><img role="img" src="https://github.com/benjamincanac.png" width="10" height="10" class="h-full w-full rounded-[inherit] object-cover" style="display: none;"><span class="font-medium text-base-500 dark:text-base-600 truncate"> </span></span>
<!--v-if--></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with b24ui correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis underline-offset-1 font-bold">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with class correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent rounded-full font-bold"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis underline-offset-1">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with default slot correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if-->Default slot</span>
<!--v-if--></span>"
`;

exports[`Badge > renders with icon correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md ps-2xs font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><i-lucide-rocket class="shrink-0 size-[15px]"></i-lucide-rocket><!--v-if--></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with label correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis underline-offset-1">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with leading and icon correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md ps-2xs font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent" leading="true"><span class="inline-flex items-center h-[17px] gap-1"><i-lucide-arrow-left class="shrink-0 size-[15px]"></i-lucide-arrow-left><!--v-if--></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with leading slot correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1">Leading slot<!--v-if--></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with leadingIcon correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent" leadingicon="i-lucide-arrow-left"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><!--v-if--></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with primary depth dark correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-base-500 dark:ring-base-600 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis underline-offset-1">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with primary depth light correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-normal text-3xs gap-1 ring ring-inset ring-base-100 dark:ring-base-150 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis underline-offset-1">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with primary depth normal correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis underline-offset-1">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with size lg correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 font-bold text-xs gap-1 rounded-lg ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[24px] gap-1"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with size md correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis underline-offset-1">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with size sm correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-4xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[16px] gap-1"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis underline-offset-1">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with size xl correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 font-bold text-md gap-1 rounded-xl ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[31px] gap-1"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with size xs correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-5xs gap-0.5 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[14px] gap-0.5"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis underline-offset-1">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with success depth dark correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-green-500 dark:ring-green-600 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis underline-offset-1">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with success depth light correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-normal text-3xs gap-1 ring ring-inset ring-green-500/17 dark:ring-green-300 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis underline-offset-1">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with success depth normal correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-green-300 dark:ring-green-330 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><span class="max-w-full whitespace-nowrap text-ellipsis underline-offset-1">Badge</span></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with trailing and icon correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><!--v-if--></span>
<!--v-if--></span>"
`;

exports[`Badge > renders with trailing slot correctly 1`] = `"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><!--v-if--></span>Trailing slot</span>"`;

exports[`Badge > renders with trailingIcon correctly 1`] = `
"<span class="select-none font-b24-secondary inline-flex items-center transition-all duration-200 ease-linear px-2 rounded-md font-bold text-3xs gap-1 ring ring-inset ring-base-150 dark:ring-base-200 text-base-800 dark:text-base-250 bg-transparent dark:bg-transparent"><span class="inline-flex items-center h-[17px] gap-1"><!--v-if--><!--v-if--></span>
<!--v-if--></span>"
`;
